src/model.rs

```diff
@@ -1,173 +1,207 @@
 use core::panic;
 use std::collections::BTreeMap;
-use std::mem::MaybeUninit;
+use std::mem::{self, MaybeUninit};
 
 use crate::constraint::Constraint;
 use crate::retcode::Retcode;
-use crate::scip_call;
 use crate::solution::Solution;
 use crate::status::Status;
 use crate::variable::{VarId, VarType, Variable};
 use crate::{ffi, scip_call_panic};
+use crate::{scip_call, scip_call_expect};
 use std::ffi::CString;
 
-#[derive(Debug, PartialEq, Eq)]
-pub enum State {
-    Created,
-    PluginsIncluded,
-    ProblemCreated,
-    Solved,
+struct ScipPtr(*mut ffi::SCIP);
+
+impl ScipPtr {
+    fn new() -> Self {
+        let mut scip_ptr = MaybeUninit::uninit();
+        scip_call_panic!(ffi::SCIPcreate(scip_ptr.as_mut_ptr()));
+        let scip_ptr = unsafe { scip_ptr.assume_init() };
+        ScipPtr(scip_ptr)
+    }
+
+    fn ptr(&self) -> *mut ffi::SCIP {
+        self.0
+    }
+}
+
+impl Default for ScipPtr {
+    fn default() -> Self {
+        Self(std::ptr::null_mut())
```

Comment:
If we had 

```suggestion
        Self::new()
```

here instead, we would have a very nice invariant that the pointer can never be null, and we could use ScipPtr structs everywhere without having to check if they are null first.

Reply:
Makes sense, thanks.

Reply:
Done, if you approve the change, please resolve the conversation.